File: src/servidor/logica.py
#---------------------------------
# Nerea Aguilar Fores
#---------------------------------

import sqlite3
from datetime import datetime

DB = "datos.db"

#Inicializar base de datos
def init_db():
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute("""
        CREATE TABLE IF NOT EXISTS datosMedidas (
            contador INTEGER PRIMARY KEY,
            tipo INTEGER,
            medicion REAL,
            fecha TEXT
        )
    """)
    conn.commit()
    conn.close()

#Añadir medidas
def anyadirMed(tipo: int, medicion: float):
    fecha = datetime.utcnow().isoformat()
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute("INSERT INTO datosMedidas (tipo, medicion, fecha) VALUES (?, ?, ?)",
              (tipo, medicion, fecha))
    conn.commit()
    conn.close()
    return {"id": c.lastrowid, "tipo": tipo, "medicion": medicion, "fecha": fecha}



#Recuperar última medida
def ultimaMed():
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute("SELECT id, tipo, medicion, fecha FROM datosMedidas ORDER BY id DESC LIMIT 1")
    fila = c.fetchone()
    conn.close()
    if fila:
        return {"id": fila[0], "tipo": fila[1], "medicion": fila[2], "fecha": fila[3]}
    return None

File: src/servidor/logica.php
<?php
//-------------------------------------------------------------
// Nerea
// Contiene la lógica de negocio. Conexión y operaciones con la BBDD
//-------------------------------------------------------------

date_default_timezone_set('Europe/Madrid');

// Inicializa la base de datos y crea la tabla si no existe
function init_db() {
    $db = new SQLite3('datos.db');
    $db->exec('CREATE TABLE IF NOT EXISTS datosMedidas (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        tipo INTEGER,
        medicion REAL,
        fecha TEXT
    )');
    return $db;
}

// Añade una nueva medida a la base de datos
// Parámetros: tipo (int), medicion (float)
// Devuelve: array con los datos insertados
function anyadirMed($tipo, $medicion) {
    $db = init_db();
    $stmt = $db->prepare('INSERT INTO datosMedidas (tipo, medicion, fecha) VALUES (:tipo, :medicion, :fecha)');
    $stmt->bindValue(':tipo', $tipo, SQLITE3_INTEGER);
    $stmt->bindValue(':medicion', $medicion, SQLITE3_FLOAT);
    $stmt->bindValue(':fecha', date('Y-m-d H:i:s'), SQLITE3_TEXT);
    $stmt->execute();

    // Corfrima el guardado
    return [
        "id" => $db->lastInsertRowID(),
        "tipo" => $tipo,
        "medicion" => $medicion,
        "fecha" => date('Y-m-d H:i:s')
    ];
}

// Consulta la última medida registrada
// Devuelve: array con la última medida o null si no hay datos
function ultimaMed() {
    $db = init_db();
    $result = $db->query('SELECT * FROM datosMedidas ORDER BY id DESC LIMIT 1');
    $row = $result->fetchArray(SQLITE3_ASSOC);
    return $row ? $row : null;
}
?>
